crates/fuel-core/src/executor.rs

```diff
@@ -625,36 +611,47 @@ where
 
         // After the execution of all transactions in production mode, we can set the final fee.
         if execution_kind == ExecutionKind::Production {
-            coinbase_tx.outputs_mut().clear();
-            coinbase_tx.outputs_mut().push(Output::coin(
-                self.config.coinbase_recipient,
-                execution_data.coinbase,
-                *self.config.consensus_parameters.base_asset_id(),
-            ));
-            block.transactions[0] = coinbase_tx.clone().into();
-        }
+            let amount_to_mint = if self.config.coinbase_recipient != ContractId::zeroed()
+            {
+                execution_data.coinbase
+            } else {
+                0
+            };
 
-        if execution_kind != ExecutionKind::DryRun {
-            coinbase_tx = self.check_coinbase(
-                block_height,
-                coinbase_tx,
-                Some(execution_data.coinbase),
-            )?;
-            self.apply_coinbase(
-                coinbase_tx,
-                block,
+            let coinbase_tx = Transaction::mint(
+                TxPointer::new(block_height, execution_data.tx_count),
+                input::contract::Contract {
+                    utxo_id: Default::default(),
+                    balance_root: Default::default(),
+                    state_root: Default::default(),
+                    tx_pointer: Default::default(),
+                    contract_id: self.config.coinbase_recipient,
+                },
+                output::contract::Contract {
+                    input_index: 0,
+                    balance_root: Default::default(),
+                    state_root: Default::default(),
+                },
+                amount_to_mint,
+                self.config.consensus_parameters.base_asset_id,
+            );
+
+            execute_transaction(
                 execution_data,
-                block_db_transaction,
+                MaybeCheckedTransaction::Transaction(coinbase_tx.into()),
             )?;
```

Comment:
Should the tx_count include the mint tx? It may be confusing later on down the road if the count gets used for something else and isn't accurate.